examples/CircleOptimisation.lean

```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2023 Siddharth Bhat. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Siddharth Bhat
+
+Riemannian Gradient Descent on S1.
+-/
+
+import SciLean
+open Lean
+open SciLean
+
+
+
+/-- choose a particular A -/
+-- I want to build the function (x, y) -> (x + y, x - y)
+-- Can I multiply it with the [1 1; -1 1] matrix? how do I do this in a way that SciLean
+-- sees that it is manifestly smooth?
+-- I took a look at `Pendulum2D.lean` which seems to not compile anymore.
+def A (x : ℝ^{2}) : ℝ^{2} := 
+  ⊞ i, 
+    let x₀ := x.get 0
+    let x₁ := x.get 1
+    let x' := #[x₀ + x₁, x₀ - x₁];
+    have H : x'.size = 2 := by simp;
+    let x' := x'.get (H ▸ i.toFin)
+    x'
```

Comment:
I already run into trouble, since I am not sure how to build the matrix

```
[1  1]
[-1 1]
``` 

and multiply it with the input vector `x` such that `SciLean` can automatically see that it is smooth. It is clear that perform index manipulation is not the way to go, but I am unsure what the correct API is.

Reply:
You can define `A` simply as
```
def A (x : ℝ^{2}) : ℝ^{2} := 
  ⊞ i, if i = 0 then x[0] + x[1] else x[0] - x[1]
```